### DISCUSSION TITLE
Ctrl+MouseClick and Ctrl+MouseDoubleClick block are messing up in fabricjs.

### DISCUSSION THREAD

**Discussion by Rudraksh3101** [fabricjs/fabric.js#8602] [General]
I am trying to implement functionality1 under if block(If Condition : Ctrl+MouseClick) and functionality2 under another if block(If condition : Ctrl+MouseDoubleClick) just one block after another but the problem I am facing is when I am trying to execute block of Ctrl+MouseDoubleClick at that time it is executing the Ctrl+MouseClick block as well which I dont want.

Could someone please help me how to differentiate?

let args = await events.awaitOneOf(
	c => events.awaitEventOnce("mouse:up", undefined, undefined, c, false),
	c => events.awaitEventOnce("mouse:dblclick", undefined, undefined, c, false));

if (args.Args?.e instanceof MouseEvent && (args.Args?.e.ctrlKey) && args.Args?.isClick) {
								console.log("In mouse single click");
							}

else if ((args.Args?.e instanceof MouseEvent && (args.Args?.e.ctrlKey) && !args.Args?.isClick)) {
							//	console.log("In mouse doble click ");
}

**Comment by Rudraksh3101** [comment]
Anyone has any idea ?

**Comment by ShaMan123** [comment]
Your ticket is not clear.
This makes it hard for the community to help and pay interest.
dblclick fires after mouseup.

**Comment by iocoker** [comment]
@Rudraksh3101 Can you explain what you're trying to do, perhaps step by step.
The question you posted is not clear at all. Also if possible create a http://jsfiddle.net/ sample code
